akd/tools/fact_check.py

```diff
@@ -0,0 +1,158 @@
+        {},
+        description="Additional logging metadata from the run.",
+    )
+
+
+class FactCheckToolConfig(BaseToolConfig):
+    """Configuration for the FactCheckTool."""
+
+    base_url: HttpUrl = Field(
+        default=HttpUrl(os.getenv("FACT_CHECK_API_URL", "http://localhost:8011")),
+        description="The base URL of the remote Fact-Checking and Correction Service.",
+    )
+    polling_interval_seconds: int = Field(
+        default=120,
+        description="How often to poll for job results.",
+    )
+    job_timeout_seconds: int = Field(
+        default=int(os.getenv("FACT_CHECK_JOB_TIMEOUT", "1800")),
+        description="Maximum time to wait for the entire job to complete (30 minutes).",
+    )
+    request_timeout_seconds: int = Field(
+        default=int(os.getenv("FACT_CHECK_REQUEST_TIMEOUT", "60")),
+        description="Timeout in seconds for each individual API request.",
+    )
+
+
+class FactCheckTool(
+    BaseTool[FactCheckInputSchema, FactCheckOutputSchema],
+):
+    """
+    A tool that calls an API to perform fact-checking on a given answer.
+    """
+
+    name = "fact_check_tool"
+    description = (
+        "Calls an API to run the FactReasoner pipeline on a question and answer."
+    )
+    input_schema = FactCheckInputSchema
+    output_schema = FactCheckOutputSchema
+    config_schema = FactCheckToolConfig
+
+    def __init__(
+        self,
+        config: FactCheckToolConfig | None = None,
+        debug: bool = False,
+    ):
+        """Initializes the FactCheckTool and its HTTP client."""
+        config = config or FactCheckToolConfig()
+        super().__init__(config, debug)
+
+        logger.info("Initializing FactCheckTool...")
+        # Set a timeout on the API requests
+        timeout = httpx.Timeout(self.config.request_timeout_seconds, connect=60.0)
+        self.api_client = httpx.AsyncClient(
+            base_url=str(self.config.base_url),
+            timeout=timeout,
+        )
+
+    async def _arun(
+        self,
+        params: FactCheckInputSchema,
+    ) -> FactCheckOutputSchema:
+        """
+        Starts a fact-checking job and polls for its completion.
+        """
+        logger.info(
+            f"Sending fact-check request for question: '{params.question[:50]}...'",
+        )
+
+        try:
+            # Start the job
+            start_response = await self.api_client.post(
+                "/fact-check/start",
+                json=params.model_dump(),
+            )
+            start_response.raise_for_status()
+            job_id = start_response.json()["job_id"]
+            logger.info(f"Successfully started job with ID: {job_id}")
+
+            # Poll for the result
+            total_wait_time = 0
+            while total_wait_time < self.config.job_timeout_seconds:
+                logger.info(f"Polling status for job {job_id}...")
+                status_response = await self.api_client.get(
+                    f"/fact-check/status/{job_id}",
```

Comment:
same as above. Probably make this configurable and default to existing endpoint.